### PR TITLE
⭐️ add internship list view

### LINKED ISSUE #619: Dashboard: Internship List view
**Describe the solution you'd like**
Implementation of the attached UI of a list view of opportunities based on code guidelines.

**Additional context**
Refer [Figma](https://www.figma.com/file/BrP5RowC7va50HyETZQUDA/Ignitus-Design?node-id=121%3A8) for User-Interface and other styles-related issue.

- You can view all the CSS, colors, typography, shadow effects, required to implement the design.
- Export the png icons from the design.

Screenshot 2020-03-12 at 3 15 38 PM

Please make this as a shared component that could be used all over the app to make the app consistent. Text is already defined in the code which should be used in this component.

### PR DESCRIPTION
**Pull request by ashishnagpal2498**
fixes #619 
1) Page can be viewed on internshipList path
2) Sharable Component - 
3) Need to change the Apply Icon - font-awesome icon not working

### CODE REVIEW

File: src/ignitus-Shared/Components/InternshipList/InternshipList.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import './style.scss';
+import * as S from './index';
+
+export default function InternshipList() {
+  return (
+    <ul style={{listStyleType: 'none', margin: '200px 30px'}}>
+      <li className="internship-list-item">
+        <div className="internship-description">
+          <div className="internship-image" />
+          <div className="description-inner">
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/InternshipList/InternshipList.js]: can be written as `descriptionContainer`

File: src/ignitus-Shared/Components/InternshipList/InternshipList.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            <S.InternshipTitle>Internship Title</S.InternshipTitle>
+            <S.SubHeading>Internship Poster | Location </S.SubHeading>
+            <S.Description>
+              But I must explain to you how all this mistaken ideas of
+              denouncing pleasure and praising pain was born and I will give you
+              a complete account of the system, and expound the actual teachings
+              of the great explorer of the truth ......
+            </S.Description>
+          </div>
+          <div className="internship-opportunity">
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/InternshipList/InternshipList.js]: can be written as `dateBtnContainer` ,` internship-opportunity` is not semantic at all.

File: src/ignitus-Shared/Components/InternshipList/index.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import styled from '@emotion/styled';
+import * as T from '../emotionStyles/shared'
+import * as C from '../emotionStyles/colors'
+
+export const InternshipTitle = styled(T.Title)`
+  color:orange
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/InternshipList/index.ts]: create a new `const` with hex code of `orange`  in `colors.js`

File: src/ignitus-Shared/Components/InternshipList/index.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+`;
+
+export const SubHeading = styled(T.Title)`
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/InternshipList/index.ts]: No need to extend `T.Title` to create `SubTitle` or `secondaryTitle`, you can create a new `const` in `shared.ts`, something like.

export const subTitle = styled.h3`
  // define font-weight & size here accordingly.
`

File: src/ignitus-Shared/Components/InternshipList/index.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+	margin-bottom: 3px;
+`
+
+export const DateInfo = styled(T.Paragraph)`
+	color: grey;
+	right: 0;
+	position: absolute;
+	top: 10px;
+	padding: 10px;
+	@media (max-width: 576px) {
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/InternshipList/index.ts]: use something like this for defining `mediaQueries` https://github.com/Ignitus/Ignitus-client/blob/master/src/ignitus-Shared/Components/emotionStyles/shared.ts#L5

File: src/ignitus-Shared/Components/InternshipList/style.scss [hunk 6]
```diff
@@ -0,0 +1,54 @@
+.internship-list-item {
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/InternshipList/style.scss]: It doesn't make sense to write a component a combination of `scss` and `emotion`, please go with with of the one approach, I would say go with `emotion`.

**Comment by divyanshu-rawat** [comment]
Branch is `onHold` until all the changes are not resolved. cc @Paarmita , @gittysachin , @debck

File: src/ignitus-Shared/Components/InternshipList/index.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+export const Description = styled(T.Paragraph)`
+	color: grey;
+	margin-bottom: 3px;
+`
+
+export const DateInfo = styled(T.Paragraph)`
+	color: grey;
+	right: 0;
+	position: absolute;
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/InternshipList/index.ts]: Please don't use `position: absolute;  right: 0; top: 10px;` I don't understand their use-case they are generally helpful in layering content over each other like if we have to use `z-index` etc.

**Comment by ashishnagpal2498** [comment]
I have redesigned the page as you suggested @divyanshu-rawat using flex. Only the icon of the Apply is left. The font awesome has is not working and blank page is shown when I put svg. I already told this to @Paarmita for other design #607 . Do let me know the changes.

File: src/ignitus-Shared/Components/InternshipList/index.ts (hunk 3 above)
Comment (ashishnagpal2498) [reply src/ignitus-Shared/Components/InternshipList/index.ts]: Okay

File: src/ignitus-Shared/Components/InternshipList/index.ts (hunk 4 above)
Comment (ashishnagpal2498) [reply src/ignitus-Shared/Components/InternshipList/index.ts]: Done

**Comment by divyanshu-rawat** [comment]
screencapture-deploy-preview-636-peaceful-nightingale-55a1b7-netlify-internshipList-2020-03-16-00_31_53

File: src/ignitus-Shared/Components/InternshipList/index.ts [hunk 8]
```diff
@@ -0,0 +1,108 @@
+import styled from '@emotion/styled';
+import * as T from '../../ignitus-DesignSystem/ignitus-Atoms/atoms';
```
Comment (ashishnagpal2498) [inline src/ignitus-Shared/Components/InternshipList/index.ts]: ignitus-DesignSystem/ignitus-Atoms/atoms , from where did you get that?

**Pull request merged**